File: service-security-kotlin-msa/api/src/main/java/org/delivery/api/domain/token/business/TokenBusiness.java
package org.delivery.api.domain.token.business;

import lombok.RequiredArgsConstructor;
import org.delivery.api.domain.token.controller.model.TokenResponse;
import org.delivery.api.domain.token.converter.TokenConverter;
import org.delivery.api.domain.token.service.TokenService;
import org.delivery.common.annotation.Business;
import org.delivery.common.error.ErrorCode;
import org.delivery.common.exception.ApiException;
import org.delivery.db.user.UserEntity;

import java.util.Optional;

@RequiredArgsConstructor
@Business
public class TokenBusiness {

    private final TokenService tokenService;
    private final TokenConverter tokenConverter;

    /**
     * 1. user entity userId 추출
     * 2. access, refresh token 발행
     * 3. converter -> token response 변경
     */
    public TokenResponse issueToken(UserEntity userEntity) {

        return Optional.ofNullable(userEntity)
                .map(UserEntity::getId)
                .map(userId -> {
                    var accessToken = tokenService.issueAccessToken(userId);
                    var refreshToken = tokenService.issueRefreshToken(userId);
                    return tokenConverter.toResponse(accessToken, refreshToken);
                })
                .orElseThrow(
                        () -> new ApiException(ErrorCode.NULL_POINT)
                );
    }

    public Long validationAccessToken(String accessToken) {
        return tokenService.validationToken(accessToken);
    }
}

File: service-security-kotlin-msa/account/src/main/kotlin/org/delivery/account/domain/token/business/TokenBusiness.kt
package org.delivery.account.domain.token.business

import org.delivery.account.common.Log
import org.delivery.account.domain.token.controller.model.TokenValidationRequest
import org.delivery.account.domain.token.controller.model.TokenValidationResponse
import org.delivery.account.domain.token.service.TokenService
import org.delivery.common.annotation.Business

@Business
class TokenBusiness(
    private val tokenService: TokenService,
) {

    companion object : Log

    fun tokenValidation(
        tokenValidationRequest: TokenValidationRequest?,
    ): TokenValidationResponse {
        //  val token = tokenValidationRequest?.tokenDto.token ??
        log.info("token validation request businuess : {}", tokenValidationRequest)
        val result = tokenService.validationToken(tokenValidationRequest?.tokenDto.toString())
        
        return TokenValidationResponse(
            userId = result
        )
    }
}